File: Script/Fattoriale_iterativo/Fattoriale.js
function fattorialeIter(n) {
  let risultato = 1;
  for (let i = 1; i <= n; i++) {
    risultato *= i;
  }
  return risultato;
}

if (process.argv.length !== 3) {
  console.log(`Uso: node ${process.argv[1]} <numero>`);
  process.exit(1);
}

const n = parseInt(process.argv[2], 10);
console.log(`Fattoriale iterativo(${n}) = ${fattorialeIter(n)}`);

File: Script/Fattoriale_iterativo/Fattoriale.java
public class Fattoriale {

    public static long fattoriale(int n) {
        long risultato = 1;
        for (int i = 1; i <= n; i++) {
            risultato *= i;
        }
        return risultato;
    }

    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Uso: java Fattoriale <numero>");
            return;
        }

        int n = Integer.parseInt(args[0]);
        System.out.println("Fattoriale iterativo(" + n + ") = " + fattoriale(n));
    }
}
